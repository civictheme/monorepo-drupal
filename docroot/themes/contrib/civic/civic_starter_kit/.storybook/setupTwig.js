const twigDrupal = require('twig-drupal-filters');

/**
 * Configures and extends a standard twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigDrupal(twig);
  return twig;
};
