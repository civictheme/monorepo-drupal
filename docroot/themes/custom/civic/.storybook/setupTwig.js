const { resolve } = require('path');
const twigDrupal = require('twig-drupal-filters');

module.exports.namespaces = {
  atoms: resolve(__dirname, '../', 'components/01-atoms'),
  molecules: resolve(__dirname, '../', 'components/02-molecules'),
  organisms: resolve(__dirname, '../', 'components/03-organisms'),
  templates: resolve(__dirname, '../', 'components/04-templates'),
};

/**
 * Configures and extends a standard twig object.
 *
 * @param {Twig} twig - twig object that should be configured and extended.
 *
 * @returns {Twig} configured twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigDrupal(twig);
  return twig;
};
