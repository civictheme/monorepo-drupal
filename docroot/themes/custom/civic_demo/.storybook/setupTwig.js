const { resolve } = require('path');
const twigDrupal = require('twig-drupal-filters');


module.exports.namespaces = {
  atoms: resolve(__dirname, '../', '.storybook-components/01-atoms'),
  molecules: resolve(__dirname, '../', '.storybook-components/02-molecules'),
  organisms: resolve(__dirname, '../', '.storybook-components/03-organisms'),
  templates: resolve(__dirname, '../', '.storybook-components/04-templates'),
};

/**
 * Configures and extends a standard twig object.
 *
 * @param {Twig} twig - twig object that should be configured and extended.
 *
 * @returns {Twig} configured twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigDrupal(twig);
  return twig;
};
