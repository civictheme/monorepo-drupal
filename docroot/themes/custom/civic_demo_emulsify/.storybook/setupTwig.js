const { resolve } = require('path');
const twigDrupal = require('twig-drupal-filters');
const twigBEM = require('bem-twig-extension');
const twigAddAttributes = require('add-attributes-twig-extension');

module.exports.namespaces = {
  atoms: [
    resolve(__dirname, '../../civic_emulsify', 'components/01-atoms'),
    resolve(__dirname, '../', 'components/01-atoms'),
  ],
  molecules: [
    resolve(__dirname, '../../civic_emulsify', 'components/02-molecules'),
    resolve(__dirname, '../', 'components/02-molecules'),
  ],
  organisms: [
    resolve(__dirname, '../../civic_emulsify', 'components/03-organisms'),
    resolve(__dirname, '../', 'components/03-organisms'),
  ],
  templates: [
    resolve(__dirname, '../../civic_emulsify', 'components/04-templates'),
    resolve(__dirname, '../', 'components/04-templates'),
  ]
};

/**
 * Configures and extends a standard twig object.
 *
 * @param {Twig} twig - twig object that should be configured and extended.
 *
 * @returns {Twig} configured twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigBEM(twig);
  twigDrupal(twig);
  twigAddAttributes(twig);
  return twig;
};
