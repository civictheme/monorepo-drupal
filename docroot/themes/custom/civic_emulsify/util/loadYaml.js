import { resolve } from 'path';
import { readFileSync } from 'fs';
import { parse } from 'yaml';
import R from 'ramda';

/**
 * Small utility function that loads a yaml file and parses it synchronously.
 * This is intended to make composition cleaner.
 *
 * @param {string} relativePath - relative path to a yaml file that will be loaded and parsed.
 *
 * @returns {string} JavaScript object that results from the yaml parsing of the specified file.
 */
export default function loadYaml(relativePath) {
  const fullPath = resolve(__dirname, relativePath);
  return parse(readFileSync(fullPath, 'utf8'));
}
