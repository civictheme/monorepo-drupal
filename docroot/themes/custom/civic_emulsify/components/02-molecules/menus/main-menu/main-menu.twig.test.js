import { join } from 'path';
import { render, Twig } from 'twig-testing-library';

import loadYaml from '../../../../util/loadYaml';
import { setupTwig, namespaces } from '../../../../.storybook/setupTwig';

setupTwig(Twig);

describe('main-menu', () => {
  it('can render an main menu', async () => {
    const { container } = await render(
      join(__dirname, 'main-menu.twig'),
      loadYaml(join(__dirname, 'main-menu.yml')),
      namespaces,
    );

    expect(container).toMatchSnapshot();
  });
});
