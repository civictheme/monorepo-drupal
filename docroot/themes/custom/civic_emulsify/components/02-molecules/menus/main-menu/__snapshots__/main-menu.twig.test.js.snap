// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main-menu can render an main menu 1`] = `
<div>
  


  <nav>
    
  
    <a
      class="toggle-expand"
      href="#"
      id="toggle-expand"
    >
      
    
      <span
        class="toggle-expand__open"
      >
        
      






        <svg
          class="toggle-expand__icon"
        >
          
      
          <use
            xlink:href="icons.svg#menu"
          />
          

        </svg>
        
      
        <span
          class="toggle-expand__text"
        >
          Main Menu
        </span>
        
    
      </span>
      
    
      <span
        class="toggle-expand__close"
      >
        
      
        <span
          class="toggle-expand__text"
        >
          Close
        </span>
        
    
      </span>
      
  
    </a>
    
  
    <div
      class="main-nav"
      id="main-nav"
    >
      
    




    
    
                      
    


      <ul
        class="main-menu"
      >
        
            


        <li
          class="main-menu__item main-menu__item--with-sub"
        >
          
      
          


          <a
            class="main-menu__link main-menu__link--with-sub"
            href="#"
          >
            
      Menu Item 1
  
          </a>
          
              
          <span
            class="expand-sub"
          />
          
          
    
                                
    


          <ul
            class="main-menu main-menu--sub main-menu--sub-1"
          >
            
            


            <li
              class="main-menu__item main-menu__item--sub main-menu__item--sub-1"
            >
              
      
          


              <a
                class="main-menu__link main-menu__link--sub main-menu__link--sub-1"
                href="#"
              >
                
      Menu Item 1 Sub 1
  
              </a>
              
          
            </li>
            
          


            <li
              class="main-menu__item main-menu__item--sub main-menu__item--sub-1 main-menu__item--with-sub"
            >
              
      
          


              <a
                class="main-menu__link main-menu__link--sub main-menu__link--sub-1 main-menu__link--with-sub"
                href="#"
              >
                
      Menu Item 1 Sub 2
  
              </a>
              
              
              <span
                class="expand-sub"
              />
              
          
    
                      
    


              <ul
                class="main-menu main-menu--sub main-menu--sub-1 main-menu--sub main-menu--sub-2"
              >
                
            


                <li
                  class="main-menu__item main-menu__item--sub main-menu__item--sub-2"
                >
                  
      
          


                  <a
                    class="main-menu__link main-menu__link--sub main-menu__link--sub-2"
                    href="#"
                  >
                    
      Menu Item 1 Sub Sub 1
  
                  </a>
                  
          
                </li>
                
          


                <li
                  class="main-menu__item main-menu__item--sub main-menu__item--sub-2"
                >
                  
      
          


                  <a
                    class="main-menu__link main-menu__link--sub main-menu__link--sub-2"
                    href="#"
                  >
                    
      Menu Item 1 Sub Sub 2
  
                  </a>
                  
          
                </li>
                
      
              </ul>
              
  
      
            </li>
            
          


            <li
              class="main-menu__item main-menu__item--sub main-menu__item--sub-1"
            >
              
      
          


              <a
                class="main-menu__link main-menu__link--sub main-menu__link--sub-1"
                href="#"
              >
                
      Menu Item 1 Sub 3
  
              </a>
              
          
            </li>
            
          


            <li
              class="main-menu__item main-menu__item--sub main-menu__item--sub-1"
            >
              
      
          


              <a
                class="main-menu__link main-menu__link--sub main-menu__link--sub-1"
                href="#"
              >
                
      Menu Item 1 Sub 4
  
              </a>
              
          
            </li>
            
      
          </ul>
          
  
      
        </li>
        
          


        <li
          class="main-menu__item main-menu__item--with-sub"
        >
          
      
          


          <a
            class="main-menu__link main-menu__link--with-sub"
            href="#"
          >
            
      Menu Item 2
  
          </a>
          
              
          <span
            class="expand-sub"
          />
          
          
    
                                
    


          <ul
            class="main-menu main-menu--sub main-menu--sub-1"
          >
            
            


            <li
              class="main-menu__item main-menu__item--sub main-menu__item--sub-1"
            >
              
      
          


              <a
                class="main-menu__link main-menu__link--sub main-menu__link--sub-1"
                href="#"
              >
                
      Menu Item 2 Sub 1
  
              </a>
              
          
            </li>
            
          


            <li
              class="main-menu__item main-menu__item--sub main-menu__item--sub-1"
            >
              
      
          


              <a
                class="main-menu__link main-menu__link--sub main-menu__link--sub-1"
                href="#"
              >
                
      Menu Item 2 Sub 2
  
              </a>
              
          
            </li>
            
          


            <li
              class="main-menu__item main-menu__item--sub main-menu__item--sub-1"
            >
              
      
          


              <a
                class="main-menu__link main-menu__link--sub main-menu__link--sub-1"
                href="#"
              >
                
      Menu Item 2 Sub 3
  
              </a>
              
          
            </li>
            
      
          </ul>
          
  
      
        </li>
        
          


        <li
          class="main-menu__item main-menu__item--with-sub"
        >
          
      
          


          <a
            class="main-menu__link main-menu__link--with-sub"
            href="#"
          >
            
      Menu Item 3
  
          </a>
          
              
          <span
            class="expand-sub"
          />
          
          
    
                                
    


          <ul
            class="main-menu main-menu--sub main-menu--sub-1"
          >
            
            


            <li
              class="main-menu__item main-menu__item--sub main-menu__item--sub-1"
            >
              
      
          


              <a
                class="main-menu__link main-menu__link--sub main-menu__link--sub-1"
                href="#"
              >
                
      Menu Item 3 Sub 1
  
              </a>
              
          
            </li>
            
          


            <li
              class="main-menu__item main-menu__item--sub main-menu__item--sub-1"
            >
              
      
          


              <a
                class="main-menu__link main-menu__link--sub main-menu__link--sub-1"
                href="#"
              >
                
      Menu Item 3 Sub 2
  
              </a>
              
          
            </li>
            
          


            <li
              class="main-menu__item main-menu__item--sub main-menu__item--sub-1"
            >
              
      
          


              <a
                class="main-menu__link main-menu__link--sub main-menu__link--sub-1"
                href="#"
              >
                
      Menu Item 3 Sub 3
  
              </a>
              
          
            </li>
            
          


            <li
              class="main-menu__item main-menu__item--sub main-menu__item--sub-1"
            >
              
      
          


              <a
                class="main-menu__link main-menu__link--sub main-menu__link--sub-1"
                href="#"
              >
                
      Menu Item 3 Sub 4
  
              </a>
              
          
            </li>
            
      
          </ul>
          
  
      
        </li>
        
      
      </ul>
      
  

    </div>
    

  </nav>
  

</div>
`;
